packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -550,6 +563,11 @@ const MetricForm: FC<MetricFormProps> = ({
             }
           }}
         >
+          {isExclusivelyForDemoDatasourceProject && (
+            <div className="alert alert-warning">
+              You are creating a metric under the demo datasource project.
```

Comment:
One thing we can do to make this even safer is prevent them from using a different datasource in this project.

So I suggest 2 changes:
1. When creating the `form`, when in the demo datasource project, set the default to the demo datasource
2. In the Data Source selector, only let them select the demo datasource

Reply:
I believe these changes mentioned here address these comments: https://github.com/growthbook/growthbook/pull/1499#issuecomment-1679612849 @lukesonnet